distribution/lib/Standard/Database/0.0.0-dev/src/JDBC/Generic_JDBC_Connection.enso

```
@@ -0,0 +1,238 @@
+from Standard.Base import all
+from Standard.Table import Table
+
+import project.Internal.Column_Fetcher as Column_Fetcher_Module
+import project.Internal.JDBC_Connection
+import project.Internal.SQL_Warning_Helper
+import project.Internal.Statement_Setter.Statement_Setter
+import project.SQL_Statement.SQL_Statement
+from project.Internal.Result_Set import read_column, result_set_to_table
+
+polyglot java import java.sql.DatabaseMetaData
+
+type Generic_JDBC_Connection
+    ## PRIVATE
+
+       A wrapper around a JDBC Connection. This can be used to access any
+       JDBC-compliant database backend, including ones that do not have full
+       support via `Connection`s. It provides schema information, and the
+       ability to execute SQL
+       queries and commands.
+
+       Unlike regular fully-supported `Connection`s, query results are returned
+       as in-memory `Table`s, rather than as `DB_Table`s.
+
+       The JDBC driver must be available on the JVM classpath.
+    private Value jdbc_connection:JDBC_Connection.JDBC_Connection
+
+    ## ADVANCED
+       GROUP Standard.Base.Input
+       ICON data_input
+
+       Connect to a database specified by a JDBC URL.
+
+       Arguments:
+       - url: The JDBC URL describing the database to connect to.
+
+       > Example
+         Connect to an H2 database instance.
+
+             connection = Generic_JDBC_Connection.connect "jdbc:h2:~/my_database"
+    connect url:Text -> Generic_JDBC_Connection =
```

Can we add a Database SPI so this would appear in Database.connect?

If the driver isn't present for the jdbc URI can we report an error telling the user where to put the jar file(s)?

---

Will do in #12101.